index.html

```diff
@@ -1,100 +1,122 @@
-    <option value="35">Okreg wyborczy 35</option>
-    <option value="36">Okreg wyborczy 36</option>
-    <option value="37">Okreg wyborczy 37</option>
-    <option value="38">Okreg wyborczy 38</option>
-    <option value="39">Okreg wyborczy 39</option>
-    <option value="40">Okreg wyborczy 40</option>
-    <option value="41">Okreg wyborczy 41</option>
-  </select>
-  <div id="wybranyokreg"></div>
-
-
-
-
-
- <!-- skrypt do odczytu plikow z excela znaleziony w necie -->
-<script src="https://cdnjs.cloudflare.com/ajax/libs/xlsx/0.17.2/xlsx.full.min.js"></script>
-
-
-<script>
-document.getElementById('przyciskexcel').addEventListener('click', function() {
-document.getElementById('filei').click();
-});
-
-document.getElementById('filei').addEventListener('change', function() {
-  var file = this.files[0];
-  if (file) {
-  loadedFile = file;
-  alert('Plik Excel został załadowany i zapisany.');
-      }
-    });
-    
-//tutaj znajduje sie kod do wczytania danych z excela 
-    document.getElementById('select').addEventListener('change', function() {
-      var selectedOption = parseInt(this.value);
-      if (loadedFile) {
-        var reader = new FileReader();
-        reader.onload = function(e) {
-          var data = new Uint8Array(e.target.result);
-          var workbook = XLSX.read(data, { type: 'array' });
-          var worksheet = workbook.Sheets[workbook.SheetNames[0]];
-          var selectedCell = worksheet[XLSX.utils.encode_cell({ r: selectedOption, c: 5 })];
-          var cellValue = selectedCell ? selectedCell.v : 'Brak danych';
-          document.getElementById('wybranyokreg').textContent = "Wybrany okreg wyborczy zawiera " + cellValue + " wyborcow";
-        };
-
-        reader.readAsArrayBuffer(loadedFile);
-      } else {
-        alert('Proszę najpierw załadować plik Excel.');
-      }
-    });
-  </script>
+    <h1>liczba wyborcow w danych okregach wyborczych</h1>
+    <button id="przyciskcsv">Załaduj plik CSV</button> <br>
+    <input type="file" id="filei" accept=".csv" style="display: none;">
+    <select id="select"></select>
+    <div id="wybranyokreg"></div>
+    <div id="kwadratodleglosci"></div>
+
+    <script>
+        var loadedFile;
+        var addedDistricts = [];
+
+        document.getElementById('przyciskcsv').addEventListener('click', function () {
+            document.getElementById('filei').click();
+        });
+
+        document.getElementById('filei').addEventListener('change', function () {
+            var file = this.files[0];
+            if (file) {
+                loadedFile = file;
+                alert('Plik CSV został załadowany i zapisany.');
+                populateDropdown();
+            }
+        });
+
+        document.getElementById('select').addEventListener('change', function () {
+            var selectedOption = this.value;
+            if (loadedFile) {
+                var reader = new FileReader();
+                reader.onload = function (e) {

```

Comment:
I see that reading the file happens 3 times in your code:
1. Here
2. In `populateDropdown`
3. In `kwadratodleglosci`

This makes the code unnecessarily complex. Notice how much overhead there is related to using `FileReader` (which is asynchronous) and then parsing the file.

The approach I suggest is as follows:

* read and parse the file after the user selects it. This means reading the file using `FileReader`, and then saving all the information from the file (the individual lines, or maybe even the individual values from each row) in a variable, so you can access it later
* when the dropdown value is changed (meaning, in this event handler), instead of reading the file again, use that already-parsed file content that you stored in a variable.

This approach means you only need to load and parse the file once. Then, after you parsed it, you can refer to the content in an easier way. This should also be faster, since you avoid re-reading the entire file when creating a new option.